metadata/src/main/java/org/apache/kafka/metadata/bootstrap/BootstrapDirectory.java

```
@@ -86,12 +84,7 @@ BootstrapMetadata readFromConfiguration() {
             return BootstrapMetadata.fromVersion(MetadataVersion.latestProduction(), "the default bootstrap");
         }
         MetadataVersion version = MetadataVersion.fromVersionString(ibp.get());
-        if (version.isLessThan(MINIMUM_BOOTSTRAP_VERSION)) {
```

@cmccabe Is it important to automatically fallback if older metadata versions are received here or can we throw an error? It's harder to fallback (but not impossible) since I removed older metadata versions.

---

the `ibp` is always empty after the config is removed by #18566

I had opened https://issues.apache.org/jira/browse/KAFKA-18740 to cleanup it. Maybe this PR can include the cleanup as well?

---

Sure, will push a commit with this change.

---

I guess this means the answer to my question is that the fallback is not important.

---

yes, we're getting rid of the fallback